controllers/configmap_controller.go

```diff
@@ -170,6 +176,43 @@ func (r *ConfigMapReconciler) reconcileServices(ctx context.Context, windowsServ
 	return nil
 }
 
+// removeOutdatedServicesConfigMaps deletes any outdated services ConfigMaps, if all nodes have moved past that version
+func (r *ConfigMapReconciler) removeOutdatedServicesConfigMaps(ctx context.Context) error {
+	watchNamespaceCMs := &core.ConfigMapList{}
+	if err := r.client.List(ctx, watchNamespaceCMs, &client.ListOptions{Namespace: r.watchNamespace}); err != nil {
+		return err
+	}
+	configMapsToDelete := []core.ConfigMap{}
+	for _, cm := range watchNamespaceCMs.Items {
+		// A possibly outdated services ConfigMap has the name "windows-services-*" where * != current WMCO version
+		if strings.HasPrefix(cm.Name, servicescm.NamePrefix) && cm.Name != servicescm.Name {
```

Comment:
Looks like the `servicescm.NamePrefix` part of the `cm.Name` slice is evaluated twice. Consider using `strings.HasPrefix()` and `strings.HasSuffix()` instead.